go-controller/pkg/ovn/master.go

```diff
@@ -370,6 +369,30 @@ func (oc *DefaultNetworkController) ensureNodeLogicalNetwork(node *kapi.Node, ho
 }
 
 func (oc *DefaultNetworkController) addNode(node *kapi.Node) ([]*net.IPNet, error) {
+	// Node subnet for the default network is allocated by cluster manager.
+	// Make sure that the node is allocated with the subnet before proceeding
+	// to create OVN Northbound resources.
+	hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node, types.DefaultNetworkName)
+	if err != nil {
+		// Log the error and return. Retry framework will keep retrying
+		// until the cluster manager has allocated the subnets.
+		klog.Infof("Failed to get node %s host subnets annotations: %v", node.Name, err)
+		return nil, err
+	}
+
+	// OVN can work in single-stack or dual-stack only.
+	currentHostSubnets := len(hostSubnets)
+	expectedHostSubnets := 1
+	// if dual-stack mode we expect one subnet per each IP family
+	if config.IPv4Mode && config.IPv6Mode {
+		expectedHostSubnets = 2
+	}
+
+	if expectedHostSubnets != currentHostSubnets {
+		klog.Errorf("Allocated Subnets %v on Node %s", hostSubnets, node.Name)
```

Comment:
Just return the error. The caller will be in charge of logging.

Reply:
Ack